NUbots/NUbook
# Upgrade dependencies

[Preview](https://deploy-preview-157--nubook.netlify.app)

Many of our current dependencies are pretty old, and we're getting lots of dependency-related warnings when running `yarn` and `yarn build`.

This PR upgrades most of our dependencies to the latest versions, including Gatsby, Webpack, React, and others. With these come several performance improvements, primarily: 

- the time it takes for `yarn dev` is down substantially, as Gatsby now builds the pages lazily on demand, and not all up front
- the time from file save in editor to updated preview in browser is now significantly shorter

Some of the upgraded packages have had a major version change with backward incompatible changes. I've fixed all the ones I noticed, but there's a possibility of something broken that I missed.

How to review:

- Check all the pages [in the preview](https://deploy-preview-157--nubook.netlify.app), click links, buttons, etc and check if anything is broken
- Clone it locally (as a fresh instance), run `yarn`, then:
  - Run `yarn dev`, make a change on one of the pages, check that it works
  - Run `yarn build`, check that it works
     - If you get this error, ignore - it's expected (it's really only an info/warning, not an error, and silencing it requires a change to an upstream dependency):

       ```
       [BABEL] Note: The code generator has deoptimised the styling of undefined as it exceeds the max of 500KB.
       ```
